backend/backend.dockerfile

```diff
@@ -1,29 +1,61 @@
-FROM python:3.7
+FROM python:3.8-slim as base
 
-LABEL maintainer="OpenStax Content Engineering"
+RUN apt-get update -qq \
+ && apt-get install -y --no-install-recommends \
+    # required by psycopg2 at build and runtime
+    libpq-dev \
+    curl \
+ && apt-get autoremove -y
 
+FROM base as builder
+
+# install poetry
+# keep this in sync with the version in pyproject.toml and Dockerfile
+ENV POETRY_VERSION 1.0.5
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH "/root/.poetry/bin:/opt/venv/bin:${PATH}"
+
+# install wait-for-it
+# allows us to wait for specific services to be up before executing scripts
 RUN curl https://raw.githubusercontent.com/vishnubob/wait-for-it/54d1f0bfeb6557adf8a3204455389d0901652242/wait-for-it.sh \
```

Comment:
wait-for-it doesn't seem to do anything within the lifetime of the `builder` stage. Should it live in the `runner` stage instead?

Reply:
I actually had it there first and then moved it up b/c i thought it would be good to just install it as part of base and not worry about it. However, you comment has me thinking that multi-stage builds are like JIT so it should go back. I'll make that change.